src/sync/commits.ts

```diff
@@ -30,23 +26,23 @@ export const getCommitTask = async (
 	jiraHost: string,
 	repository: Repository,
 	cursor?: string | number,
-	perPage?: number): Promise<CommitData> => {
-	const itemNumber = Number(cursor?.toString().split(" ").pop() || 0);
-	const itemLimit = await numberFlag(NumberFlags.SYNC_MAIN_COMMIT_AMOUNT, 99999999999, jiraHost);
-	if (itemNumber >= itemLimit) {
-		logger.info({itemNumber, itemLimit}, "Reached main commit amount needed, stopping commit sync")
-		return {
-			edges: [],
-			jiraPayload: undefined
-		};
-	}
+	perPage?: number): Promise<TaskPayload<CommitQueryNode, JiraCommitData>> => {
+
 	logger.debug("Syncing commits: started");
 	const { edges, commits } = await fetchCommits(gitHubClient, repository, cursor, perPage);
 	const jiraPayload = await transformCommit({ commits, repository });
 	logger.debug("Syncing commits: finished");
 
+	const timeCutoff = Date.now() - await numberFlag(NumberFlags.SYNC_MAIN_COMMIT_TIME_LIMIT, Date.now(), jiraHost);
```

Comment:
I'd like to suggest move the `timeCutoff` to close where they are used, which is inside the `if(edges?.length)`

Reply:
This is minor I assume: Will the task be executed in multiple loop/async-subtasks/etc ( I don't know how to describe, lol) <-- but I mean, each time the task is executed, the `await numberFlag` is called. Not sure how does this work, but will it coz some http traffic and performance issue?

Reply:
launchdarkly doesn't call the API every time we request the FF.  From my understanding it caches the results.  The initial wait time is a bit longer to call the API, but every subsequent call is then using the cached value which is essentially instantaneous.